src/index.tsx

```diff
@@ -23,22 +23,124 @@ async function main(GitURL: string, branchName?: string, folderPath?: string) {
     await $`git checkout ${branchName}`;
 
     await fs.remove(path.join(tempFolder, '.git'));
-    await fs.copy(
-        folderPath ? path.join(tempFolder, folderPath) : tempFolder,
-        targetFolder,
-        { overwrite: true }
-    );
+    
+    const sourceFolder = folderPath ? path.join(tempFolder, folderPath) : tempFolder;
+    
+    // Check if source is a file or directory
+    const sourceStat = await fs.stat(sourceFolder);
+    
+    if (sourceStat.isFile()) {
+        // If it's a file, copy it directly to target directory with same name
+        const fileName = path.basename(sourceFolder);
+        await fs.copy(sourceFolder, path.join(targetFolder, fileName), { overwrite: true });
+    } else {
+        // If it's a directory, copy contents to target directory
+        await fs.copy(sourceFolder, targetFolder, { overwrite: true });
+    }
+}
+
+async function removeSubmodule(submodulePath?: string) {
+    const currentDir = process.cwd();
+    
+    // If no path provided, list current submodules
+    if (!submodulePath) {
+        try {
+            const result = await $`git submodule status`;
+            console.log('Current submodules:');
+            console.log(result.stdout);
+            console.log('\nUsage: xgit submodule remove <path>');
+            return;
+        } catch (error) {
+            console.error('Error: This does not appear to be a git repository with submodules.');
+            process.exit(1);
+        }
+    }
+
+    try {
+        // Check if we're in a git repository
+        await $`git rev-parse --git-dir`;
+        
+        // Check if submodule exists
+        try {
+            await $`git submodule status ${submodulePath}`;
+        } catch (error) {
+            console.error(`Error: Submodule '${submodulePath}' not found.`);
+            process.exit(1);
+        }
+        
+        console.log(`Removing submodule: ${submodulePath}`);
+        
+        // Remove submodule entry from .gitmodules (if .gitmodules exists)
+        try {
+            await $`git config --file .gitmodules --remove-section submodule.${submodulePath}`;
+        } catch (error) {
+            console.warn(`Warning: Could not remove from .gitmodules (file may not exist or section not found)`);
+        }
+        
+        // Remove submodule entry from .git/config
+        try {
+            await $`git config --remove-section submodule.${submodulePath}`;
+        } catch (error) {
+            console.warn(`Warning: Could not remove from .git/config (section may not exist)`);
+        }
+        
+        // Remove from git index
+        await $`git rm --cached ${submodulePath}`;
+        
+        // Remove the submodule directory if it exists
+        const submoduleDir = path.join(currentDir, submodulePath);
+        if (await fs.pathExists(submoduleDir)) {
+            await fs.remove(submoduleDir);
+            console.log(`Removed directory: ${submodulePath}`);
+        }
```

Comment:
```suggestion
            await fs.remove(submodulePath);
```
子模块本身就是个目录，直接删除即可，有什么问题，对应 API 自然会报错。